internal/service/dataexchange/event_action.go

```diff
@@ -0,0 +1,416 @@
+			return nil, resDiag
+		}
+		input.Event.RevisionPublished = &awstypes.RevisionPublished{
+			DataSetId: eventRevisionPublished.DataSetId.ValueStringPointer(),
+		}
+	}
+
+	return &input, resDiag
+}
+
+func (r *resourceEventAction) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	conn := r.Meta().DataExchangeClient(ctx)
+
+	var state resourceEventActionModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := findEventActionByID(ctx, conn, state.ID.ValueString())
+	if tfresource.NotFound(err) {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataExchange, create.ErrActionSetting, ResNameEventAction, state.ID.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(r.buildModelFromOutput(ctx, out, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *resourceEventAction) buildModelFromOutput(ctx context.Context, output *dataexchange.GetEventActionOutput, data *resourceEventActionModel) diag.Diagnostics {
+	resDiag := diag.Diagnostics{}
+	data.ARN = types.StringPointerValue(output.Arn)
+
+	eventObjectValue, diags := fwtypes.NewObjectValueOf[eventRevisionPublishedModel](ctx, &eventRevisionPublishedModel{
+		DataSetId: types.StringPointerValue(output.Event.RevisionPublished.DataSetId),
+	})
+	if diags.HasError() {
+		resDiag.Append(diags...)
+		return resDiag
+	}
+
+	data.EventRevisionPublished = eventObjectValue
+
+	model := actionExportRevisionToS3Model{
+		Bucket:     types.StringPointerValue(output.Action.ExportRevisionToS3.RevisionDestination.Bucket),
+		KeyPattern: types.StringPointerValue(output.Action.ExportRevisionToS3.RevisionDestination.KeyPattern),
+	}
+	if output.Action.ExportRevisionToS3.Encryption != nil {
+		model.S3EncryptionType = fwtypes.StringEnumValue(output.Action.ExportRevisionToS3.Encryption.Type)
+		model.S3EncryptionKmsKeyArn = types.StringPointerValue(output.Action.ExportRevisionToS3.Encryption.KmsKeyArn)
+	}
+
+	actionObjectValue, diags := fwtypes.NewObjectValueOf[actionExportRevisionToS3Model](ctx, &model)
+	if diags.HasError() {
+		resDiag.Append(diags...)
+		return resDiag
+	}
+
+	data.ActionExportRevisionToS3 = actionObjectValue
+
+	return resDiag
+}
+
+func (r *resourceEventAction) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	conn := r.Meta().DataExchangeClient(ctx)
+
+	var data, state resourceEventActionModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !data.EventRevisionPublished.Equal(state.EventRevisionPublished) {
+		_, err := conn.DeleteEventAction(ctx, &dataexchange.DeleteEventActionInput{
+			EventActionId: state.ID.ValueStringPointer(),
+		})
```

Comment:
In most cases, the Update method should not directly call a Delete API. Instead, the attribute that triggers a replacement should be marked with a `RequiresReplace()` plan modifier.
https://developer.hashicorp.com/terraform/plugin/framework/resources/plan-modification#requiresreplace